### ISSUE TITLE
formatter callback function, how to do it?

### ISSUE THREAD

**Issue by realbugger** [kirjs/react-highcharts#121]
Based on my limited experience, it looks like most of work for create a highchart is about specifying the `config` object (which by itself is pretty powerful). But I haven't figured out a way to write formatter callback function. Obviously I can't access the chart data with `this.y`. What would be the right way to define my own formatter function? 

Thanks.

**Comment by kirjs** [comment]
How would you do it with just `Highcharts`?

**Comment by realbugger** [comment]
I guess the part I didn't get is that, in `Highcharts`, when a chart is created, one can do 
`var chart = new Highcharts.Chart(options)`. Within `options`, one can always refer to the chart object with `this`, therefore the callback function can access chart datapoint with something like `this.point` or `this.y` (example below is from this [JSFiddle](http://jsfiddle.net/JVNjs/320/)).
```
formatter:function() {
  var pcnt = (this.y / dataSum) * 100;
  return Highcharts.numberFormat(pcnt) + '%';
}
```
But in `react-highcharts`, how can I get the chart object handler so that I can refer to the datapoints like the example above?

**Comment by kirjs** [comment]
Would this help?
https://github.com/kirjs/react-highcharts#accessing-highcharts-api-after-render

**Comment by realbugger** [comment]
Thanks for you prompt reply, @kirjs. Yes I did look into that. I could access the chart instance via `ref` without problem.
```
<ReactHighcharts
  config = {chartConfig}
  ref={(ref) => {if (ref) console.log(ref.getChart());}}
/>);
```
I would assume that `ref.getChart()` would give me the same handler as the `this` in `this.y` (as in this [JSFiddle](http://jsfiddle.net/JVNjs/320/)), therefore I should be able to access `this.y` in formatter function with `ref.getChart().y`. But, nope, if I `console.log(ref.getChart().y)`, it shows it's undefined.

**Comment by kirjs** [comment]
Can't you just pass it as a part of the config object?

**Comment by realbugger** [comment]
@kirjs you're right. All I need is to pass it as part of the config object. Actually that's the first thing I did. However, it turned out that I mistakenly used ES6 arrow function to define formatter callback function, where `this` means differently than defining it with `function` keyword.

This was how I did it initially.
```
...
// Wrong. 'this' here refers to the current react component, this.y is undefined
formatter: () => this.y + ' %' 
...
```
Once I change to the vanilla function definition, problem solved.
```
...
// Correct. 'this' here refers to the data point 
formatter: function() {return this.y + ' %';} 
...
```
I never paid much attention to the nuance of arrow function. Now lessons learned. Closing the thread.
